### PR TITLE
Add ServiceMonitor installation

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#27557, giantswarm/giantswarm#27145 and giantswarm/roadmap#2567

This PR adds ServiceMonitor definition from upstream and configures it. 

@giantswarm/team-atlas can you take a look at the relabelings? Thanks <3

### Checklisto

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by ced0ps — APPROVED** [review]

**Review by ced0ps — APPROVED** [review]
Other than the relabeling LGTM

**Comment by ubergesundheit** [comment]
I think we also need to check if we can add the `ip` label to the metrics. Related https://github.com/giantswarm/prometheus-rules/pull/554/files

**Comment by hervenicol** [comment]
Concerning the relabelling, "there's a hole in the racket" as we say in French :rofl: 

We used to enforce relabellings with this kyverno policy: https://github.com/giantswarm/kyverno-policies-observability/blob/main/policies/observability/ServiceMonitor.yaml
As you can see, that makes a lot of relabellings.
Here is the result for the `grafana` servicemonitor on `gauss` for instance:
```
    relabelings:
    - action: replace
      replacement: gauss
      targetLabel: cluster_id
    - action: replace
      replacement: management_cluster
      targetLabel: cluster_type
    - action: replace
      replacement: highest
      targetLabel: service_priority
    - action: replace
      replacement: aws
      targetLabel: provider
    - action: replace
      replacement: gauss
      targetLabel: installation
    - action: replace
      regex: ;(.*)
      replacement: $1
      separator: ;
      sourceLabels:
      - namespace
      - __meta_kubernetes_namespace
      targetLabel: namespace
    - action: replace
      sourceLabels:
      - __meta_kubernetes_pod_label_app_kubernetes_io_name
      targetLabel: app
    - action: replace
      sourceLabels:
      - __meta_kubernetes_pod_label_app_kubernetes_io_instance
      targetLabel: instance
    - action: replace
      sourceLabels:
      - __meta_kubernetes_pod_name
      targetLabel: pod
    - action: replace
      sourceLabels:
      - __meta_kubernetes_pod_container_name
      targetLabel: container
    - action: replace
      sourceLabels:
      - __meta_kubernetes_pod_node_name
      targetLabel: node
    - action: replace
      sourceLabels:
      - __meta_kubernetes_node_label_role
      targetLabel: role
    - action: replace
      replacement: giantswarm
      targetLabel: customer
    - action: replace
      replacement: monitoring
      targetLabel: organization
    - action: replace
      regex: org-(.*)
      replacement: ${1}
      sourceLabels:
      - organization
      targetLabel: organization
```


BUT:
* the policy is not deployed to WCs
* current version won't work in WCs
* even on MCs it's not applied in `kube-system` namespace

So, we (Atlas) have to decide what we want to do.
Basically:
* either have some documentation listing expected relabellings, and get rid of this policy
* or fix our observability-policy

Personally, I just discovered (this week) this issue, and everyday I discover that it's worse than I thought :see_no_evil:  so I don't have an answer yet.

**Comment by ubergesundheit** [comment]
Thank you for the comment @hervenicol 

I think I'll wait for stuff to settle before continuing with this.

**Comment by ced0ps** [comment]
This has been implemented in to the most recent release of cert-manager
